File: testng/src/main/java/com/shoppingapp/Billing.java
package com.shoppingapp;

public class Billing {
    private double totalAmount;
    private double discount;

    public void setTotalAmount(double totalAmount) {
        this.totalAmount = totalAmount;
    }
    

    public void applyDiscount(double discount) {
        this.discount = discount;
    }

    public double calculateFinalAmount() {
        return totalAmount - (totalAmount * discount / 100);
    }

    public boolean validatePayment(String cardNumber) {
        return cardNumber.matches("\\d{16}"); // Simplified validation
    }
}

File: behave/src/main/shoppingapp/billing.py
class Billing:
    def __init__(self):
        self.total_amount = 0
        self.discount = 0

    def set_total_amount(self, total_amount):
        self.total_amount = total_amount

    def apply_discount(self, discount):
        self.discount = discount

    def calculate_final_amount(self):
        return self.total_amount - (self.total_amount * self.discount / 100)

    def validate_payment(self, card_number):
        return len(card_number) == 16 and card_number.isdigit()
